atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -235,6 +236,7 @@ public static SerializableTransactionManager create(
             LockServerOptions lockServerOptions,
             boolean allowHiddenTableAccess,
             Class<?> callingClass) {
+        validateRuntimeConfig(runtimeConfig.get());
```

Comment:
nit: can make the guarantees easier to enforce by pushing the validation method down into `create`.